File: crates/domain-scan-core/tests/fixtures/typescript/schemas.ts
// Test fixture: TypeScript schemas (Zod, Effect Schema, Drizzle)

import { z } from 'zod';
import { Schema } from '@effect/schema';
import { pgTable, serial, varchar, integer } from 'drizzle-orm/pg-core';

// Zod schema
export const UserSchema = z.object({
  name: z.string(),
  email: z.string().email(),
  age: z.number().optional(),
});

// Effect Schema
export const ProductSchema = Schema.Struct({
  id: Schema.String,
  title: Schema.String,
  price: Schema.Number,
});

// Drizzle table
export const users = pgTable('users', {
  id: serial('id').primaryKey(),
  name: varchar('name', { length: 255 }),
  email: varchar('email', { length: 255 }).unique(),
  age: integer('age'),
});

File: crates/domain-scan-core/tests/fixtures/python/schemas.py
from pydantic import BaseModel
from dataclasses import dataclass
from typing import Optional, TypedDict


class UserSchema(BaseModel):
    id: int
    name: str
    email: Optional[str] = None


class CreateUserRequest(BaseModel):
    name: str
    email: str
    age: Optional[int] = None


@dataclass
class UserDTO:
    id: int
    name: str
    email: str = ""


class UserDict(TypedDict):
    id: int
    name: str
    email: str
